internal/generator/vector/output/otlp/transform.go

```diff
@@ -140,15 +164,19 @@ o = {
 	BackwardCompatBaseResourceAttributes = `
 # Append backward compatibility attributes
 resource.attributes = append( resource.attributes,
-	[{"key": "log_type", "value": {"stringValue": .log_type}}]
+	[
+      {"key": "log_type", "value": {"stringValue": .log_type}},
+      {"key": "log_source", "value": {"stringValue": .log_source}},
+      {"key": "openshift.cluster_id", "value": {"stringValue": .openshift.cluster_id}}
+    ]
 )
 `
 	BackwardCompatContainerResourceAttributes = `
 # Append backward compatibility attributes for container logs
 resource.attributes = append( resource.attributes,
-	[{"key": "kubernetes_pod_name", "value": {"stringValue": get!(.,["kubernetes","pod_name"])}},
-	{"key": "kubernetes_container_name", "value": {"stringValue": get!(.,["kubernetes","container_name"])}},
-	{"key": "kubernetes_namespace_name", "value": {"stringValue": get!(.,["kubernetes","namespace_name"])}}]
+	[{"key": "kubernetes.pod_name", "value": {"stringValue": get!(.,["kubernetes","pod_name"])}},
+	{"key": "kubernetes.container_name", "value": {"stringValue": get!(.,["kubernetes","container_name"])}},
+	{"key": "kubernetes.namespace_name", "value": {"stringValue": get!(.,["kubernetes","namespace_name"])}}]
```

Comment:
Do you know why this becomes `kubernetes_namespace_name` in the final configuration? (see for example `otlp_all.toml`)
This also applies to the other attributes and the result is that Loki currently does not pick them up as it watches for `kubernetes.namespace_name`.

Reply:
Sanitizing by de-dotting (de-slash, de-dash) occurs on the entire list of LabelKeys and/or DefaultLabelKeys and I was a compatibility issue fix from fluentd.   https://github.com/openshift/cluster-logging-operator/blob/master/internal/generator/vector/output/loki/loki.go#L187

It's been that way in the config since original vector fix.

Reply:
Yes, that's true. But we're still on the OTLP side here, which can (and does) use dots. The label will be called `kubernetes_namespace_name` on the Loki side, but it's able to pull the information from an attribute called `kubernetes.namespace_name`.

Reply:
It's also possible that the file in the unit-test does not reflect the reality yet? While I did not see the correct information on Friday, it seems to work now connecting a CLO with this PR to a LokiStack. I wonder why it did not work on Friday.

Reply:
I am just now updating unit tests for this PR so I will leave this comment open for now